### PR TITLE
Fix #1981: Implement JDK8 methods of java.lang.Long.

### LINKED ISSUE #1981: Methods of java.lang.Long from JDK 8 are missing
* `BYTES`
* `toUnsignedString(i, radix)`
* `toUnsignedString(i)`
* `parseUnsignedLong(s, radix)`
* `parseUnsignedLong(s)`
* `hashCode(value)`
* `compareUnsigned(x, y)`
* `divideUnsigned(dividend, divisor)`
* `remainderUnsigned(dividend, divisor)`
* `sum(a, b)`
* `max(a, b)`
* `min(a, b)`

FTR, similar methods were implemented for `Integer` in d03c276b6bc23315ee0d4ca9644435037805b05f

### PR DESCRIPTION
**Pull request by sjrd**
* BYTES
* toUnsignedString(i, radix)
* toUnsignedString(i)
* parseUnsignedLong(s, radix)
* parseUnsignedLong(s)
* hashCode(value)
* compareUnsigned(x, y)
* divideUnsigned(dividend, divisor)
* remainderUnsigned(dividend, divisor)
* sum(a, b)
* max(a, b)
* min(a, b)

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: javalanglib/src/main/scala/java/lang/Long.scala [hunk 1]
```diff
@@ -43,123 +43,318 @@ object Long {
...
 
-    if (s.isEmpty || radix < Character.MIN_RADIX || radix > Character.MAX_RADIX) {
-      fail()
-    } else if (s.charAt(0) == '-') {
-      -parseLong(s.substring(1), radix)
+    val unsignedResult = parseUnsignedLongInternal(s, radix, start)
+
+    if (neg) {
+      val result = -unsignedResult
+      if (result > 0)
```
Comment (nicolasstucki) [inline javalanglib/src/main/scala/java/lang/Long.scala]: This sould be `if (result >= 0)`.

Comment (nicolasstucki) [inline javalanglib/src/main/scala/java/lang/Long.scala]: Oh, no. I see why now, my mistake.

**Comment by nicolasstucki** [comment]
LGTM

**Comment by sjrd** [comment]
Rebased (no conflict).

**Pull request merged**